src/bidiMapper/domains/network/NetworkProcessor.ts

```diff
@@ -296,6 +295,40 @@ export class NetworkProcessor {
             return urlPattern;
           }
 
+          if (urlPattern.protocol) {
+            urlPattern.protocol = unescapeURLPattern(urlPattern.protocol);
+            if (!urlPattern.protocol.match(/^[a-zA-Z+-.]+$/)) {
+              throw new InvalidArgumentException('Forbidden characters');
+            }
+          }
+          if (urlPattern.hostname) {
+            urlPattern.hostname = unescapeURLPattern(urlPattern.hostname);
+          }
+          if (urlPattern.port) {
+            urlPattern.port = unescapeURLPattern(urlPattern.port);
+          }
+          if (urlPattern.pathname) {
+            urlPattern.pathname = unescapeURLPattern(urlPattern.pathname);
+            if (urlPattern.pathname[0] !== '/') {
+              urlPattern.search = `/${urlPattern.search}`;
+            }
+            if (
```

Comment:
Nit: Let's move the check that that may throw first.

Reply:
That would not be according to the spec

Reply:
Like I said nit. But if we throw anyway we should not do extra work - tldr spec is wrong.